webapp/channels/src/components/integrations/add_incoming_webhook/add_incoming_webhook.tsx

```diff
@@ -43,45 +43,38 @@ type Props = {
     };
 };
 
-type State = {
-    serverError: string;
-};
-
-export default class AddIncomingWebhook extends React.PureComponent<Props, State> {
-    constructor(props: Props) {
-        super(props);
-
-        this.state = {
-            serverError: '',
-        };
-    }
+const AddIncomingWebhook = ({
+    team,
+    enablePostUsernameOverride,
+    enablePostIconOverride,
+    actions,
+}: Props) => {
+    const [serverError, setServerError] = useState('');
 
-    addIncomingHook = async (hook: IncomingWebhook) => {
-        this.setState({serverError: ''});
+    const addIncomingHook = async (hook: IncomingWebhook) => {
```

Comment:
Let's use a `useCallback` so we don't pass a new function on every render.

Reply:
Done